core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -961,6 +961,14 @@ public XContentBuilder dateValueField(XContentBuilderString rawFieldName, XConte
         return this;
     }
 
+    public XContentBuilder timeValueField(String rawFieldName, String readableFieldName, TimeValue timeValue) throws IOException {
```

Comment:
(or reverse them, so that a new `TimeValue` is only constructed if a readable rawTime is used)